Include exit code in report for `execExternalProgram`

zerbina opened the pull request:
The `execution of external program failed` error messages now
contain the exit code again, instead of always showing '0'

Fixes https://github.com/nim-works/nimskull/issues/205

---
<!-- Note: section break (`---`) onwards is not in CI merge commit -->

## Notes for Reviewers
* I tried adding a test for the message using `cmd: "nim r $file"`, but this doesn't work, as `nimout` is compared on a per-line basis and the the external program name is platform specific

alaviss commented:
I suppose this would fix #205? If so can you add that to the commit message/PR description?

zerbina committed:
Include exit code in report for `execExternalProgram`

The `execution of external program failed` error messages now
contain the exit code again, instead of always showing '0'

Fixes https://github.com/nim-works/nimskull/issues/205

haxscramper reviewed (approved):
Yes, I think that would fix #205

zerbina commented:
Sorry, I forgot that there was an issue filed for this already :P. Both the PR and commit message body now include a `Fixes` for the issue

bors r+
